modules/benchmark/R/clone_run.R

```diff
@@ -0,0 +1,115 @@
+##' @param settings settings or multisettings object 
+##' @param model model id 
+##' @param con database connection
+##' @export 
+##' 
+##' @author Betsy Cowdery 
+##' @importFrom dplyr tbl filter rename collect select 
+
+clone_run <- function(settings, model, con){
+  
+  # For testing: 
+  # bety <- betyConnect("web/config.php")
+  # con <- bety$con
+  # settings <- read.settings("/fs/data2/output/PEcAn_1000008371/pecan.CHECKED.xml")
+  # model = 1000000020
+  
+  settings.old <- settings 
+  settings.new <- list()
+  
+  # Copy over settings that won't change
+  settings.new$meta.analysis <- settings.old$meta.analysis
+  settings.new$ensemble <- settings.old$ensemble
+  settings.new$run$site <- settings.old$run$site
+  settings.new$run$start.date <- settings.old$run$start.date
+  settings.new$run$end.date <- settings.old$run$end.date
+  
+  # New model info
+  model.new <- tbl(bety, "models") %>% filter(model == id) %>% collect()
+  # List of all/required inputs for the new model 
+  inputs_all <- tbl(bety, "modeltypes_formats") %>% filter(model.new$modeltype_id == modeltype_id) %>% select(one_of("tag", "required")) %>% collect
+  inputs_req <- inputs_all$tag[inputs_all$required]
+  
+  #------ Update model ------#
+  
+  settings.new$model$id <- model.new$id
+  
+  #------ Update inputs ------#
+  
+  for(tag in names(settings.old$run$inputs)){
+    
+    # Check that the given input is required for the new model
+    if(tag %in% inputs_all$tag){
+      
+      # Check lookup table to see what format input needs to be in 
+      # For example for met we need the CF standard 
+      # For IC it depends on pool vs cohort - may need special cases
+      tags_std <- read.csv(
+        system.file("clone_run_inputs_lookup.csv", package = "PEcAn.benchmark"), 
+        stringsAsFactors = FALSE) 
+      new_format <- c()
+      
+      # Check if old model input is in the correct format
+      old.input <- tbl(bety, "inputs") %>% 
+        filter(settings.old$run$inputs[[tag]]$id == id) %>% 
+        select( one_of("id", "format_id", "parent_id")) %>% collect()
+      
+      if(!is.na(old.input$parent_id)){
+        parent.input <- tbl(bety, "inputs") %>% 
+          filter(old.input$parent_id == id) %>% 
+          select( one_of("id", "format_id", "parent_id")) %>% collect()
+      }else{
+        parent.input <- list(format_id = NA)
+      }
+      
+      if(old.input$format_id == new_format){
+        # The input id points to a file that is already in the right format to be converted
+        settings.new$run$inputs[[tag]]$id  <- old.input$id 
+      }else if(parent.input$format_id == new_format){
+        # The parent id points to a file that is in the right format to be converted
+        settings.new$run$inputs[[tag]]$id  <- parent.input$id 
+      }else if(is.na(old.input$parent_id)){
+        # If the input has no parent id, assume it's raw data (is that safe???)
+        settings.new$run$inputs[[tag]]$id  <- old.input$id
```

Comment:
Not safe. An input could be in a model specific format that was directly inserted into inputs. This is becoming less common for things with a standard (e.g. met) but will be very common for things where there's a model-specific default database (e.g. soils in ED and CLM)